### DISCUSSION TITLE
Admin as `npm` package

### DISCUSSION THREAD

**Discussion by olivermrbl** [medusajs/medusa#2829] [Feature Requests]
### What
Shipping Medusa Admin as a core plugin through `npm` to allow for versioning

### Motivation
Medusa Admin is currently a separate entity from the core. More often than not, this is frustrating for our users, as they may experience issues due to a version mismatch between Medusa Admin and the core package.  

Right now, if you want to avoid upgrading admin, you'll have to dig out the specific commit that is compatible with your version of Medusa. This is not ideal and generally a bad developer experience.

To address these issues, we are bringing the admin closer to the core project by deprecating the separate admin repository and moving the admin to our monorepo. This allows us to version Medusa Admin similarly to all other packages. Going forward, you'll be able to match your core Medusa version with a specific Medusa Admin version ensuring compatibility.

**Comment by mehmetuken** [comment]
https://github.com/directus/directus/blob/bb176c44fb06c7f95da6678f319d4575ddaea97b/api/src/app.ts#L182

Hi, this is how directus does it. maybe it will help while implementing.

**Comment by FaZeRs** [comment]
Will this package be extendable? In general I want to know if it will be possible to create custom pages and override existing components?

**Reply by srindom** [reply]
This will be part of the second iteration on the project. Initially, we are only moving the admin dashboard to NPM.

**Comment by darkcherrycreative** [comment]
I've seen that this has now been completed under #3334, do we know when this feature will be available?

**Reply by olivermrbl** [reply]
This will be part of 1.8 coming out later this month or early April. Publishing pre-release this week :)

**Reply by darkcherrycreative** [reply]
> This will be part of 1.8 coming out later this month or early April. Publishing pre-release this week :)

Excellent, this will be a real leap forwards in deployment - will keep an eye out.

**Reply by olivermrbl** [reply]
@darkcherrycreative You can find the plugin in the latest pre-release. I would love if you could give it a spin and share your thoughts around it :)

**Comment by tunchunairarko** [comment]
When can we expect to be able to extend/customize the admin panel?

**Reply by ripvannwinkler** [reply]
Echoing the concerns of @tunchunairarko and others, the admin package needs extensibility. I recently stumbled on Medusa and wanted to give it a serious look, but I have some atypical requirements that are not satisfied by the built-in functionality (e.g., item limits per user, customer group budgets).

**Reply by olivermrbl** [reply]
@tunchunairarko @ripvannwinkler – The feature is under development. We've recently reached a proof-of-concept state, where we are successfully injecting custom components into admin from a Medusa project. It is important to note, these customizations are added without ejecting the project, thereby keeping the versioning capabilities in tact.

We are happy with the direction we are moving, but there's still a lot of work todo and things to figure out before a beta release. Especially the constraints of the feature are important to thoroughly think through. We want to provide as much extensibility as possible, but without jeopardizing the core functionality of the admin system. 

You were asking for a timeline, but unfortunately, I can't provide any guarantees here. We hope to share a beta version within the current quarter.

**Comment by ripvannwinkler** [comment]
Totally understood Oliver. It's nice to hear there's progress.

**Comment by tabilzad** [comment]
@olivermrbl So the admin UI will now have to be exposed to public? Is there any way to configure the admin app to be served on another host, separate from backend?
